Add-modern-config-to-template-light (#37953)
home-assistant/home-assistant.github.io: source/_integrations/template.markdown

      required: false
      type: boolean
      default: true
light:
  description: List of your lights.
  required: true
  type: map
  keys:
    level:
      description: Defines a template to get the brightness of the light.
      required: false
      type: template
      default: optimistic
    temperature:
      description: Defines a template to get the color temperature of the light.
      required: false
      type: template
      default: optimistic
    hs:

>       description: Defines a template to get the HS color of the light. Must render   tuple (hue, saturation).

Reviewer: ```suggestion
      description: Defines a template to get the HS color of the light. Must render a tuple (hue, saturation).
```

      required: false
      type: template
      default: optimistic
    rgb:

>       description: Defines a template to get the RGB color of the light. Must render   tuple or a list (red, green, blue).

Reviewer: ```suggestion
      description: Defines a template to get the RGB color of the light. Must render a tuple or a list (red, green, blue).
```

      required: false
      type: template
      default: optimistic
    rgbw:

>       description: Defines a template to get the RGBW color of the light. Must render   tuple or a list (red, green, blue, white).

Reviewer: ```suggestion
      description: Defines a template to get the RGBW color of the light. Must render a tuple or a list (red, green, blue, white).
```

      required: false
      type: template
      default: optimistic
    rgbww:
      description: Defines a template to get the RGBWW color of the light. Must render a tuple or a list (red, green, blue, cold white, warm white).
      required: false
      type: template
      default: optimistic
    supports_transition:

>       description: Defines a template to get if light supports transition. Should  return boolean value (True/False). If this value is `True` transition parameter  in a turn on or turn off call will be passed as a named parameter `transition` t  either of the scripts.

Reviewer: ```suggestion
      description: Defines a template to get if the light supports transition. Should return a boolean value (True/False). If this value is `True`, the transition parameter in a `turn on` or `turn off` call will be passed as a named parameter `transition` in either of the scripts.
```

      required: false
      type: template
      default: false
    effect_list:

>       description: Defines a template to get the list of supported effects. Must render  a list

Reviewer: ```suggestion
      description: Defines a template to get the list of supported effects. Must render a list.
```

      required: inclusive
      type: template
      default: optimistic
    effect:
      description: Defines a template to get the effect of the light.
      required: inclusive
      type: template
      default: optimistic
    min_mireds:

>       description: Defines a template to get the min mireds value of the light.

Reviewer: ```suggestion
      description: Defines a template to get the minimum mired value of the light.
```

      required: false
      type: template
      default: optimistic
    max_mireds:

>       description: Defines a template to get the max mireds value of the light.

Reviewer: ```suggestion
      description: Defines a template to get the maximum mired value of the light.
```

      required: false
      type: template
      default: optimistic
    turn_on:

>       description: Defines an action to run when the light is turned on. May receive  variables `brightness` and/or `transition`.

Reviewer: ```suggestion
      description: Defines an action to run when the light is turned on. May receive the variables `brightness` and/or `transition`.
```

      required: true
      type: action
    turn_off:

>       description: Defines an action to run when the light is turned off. May receive  variable `transition`.

Reviewer: ```suggestion
      description: Defines an action to run when the light is turned off. May receive the variable `transition`.
```

      required: true
      type: action
    set_level:

>       description: Defines an action to run when the light is given a brightness command. The script will only be called if the `turn_on` call only ha  brightness, and optionally transition. Receives variables `brightness` an  optionally `transition`.

Reviewer: ```suggestion
      description: Defines an action to run when the light is given a brightness command. The script will only be called if the `turn_on` call only ha brightness, and optionally transition. Receives variables `brightness` and, optionally, `transition`.
```

      required: false
      type: action
    set_temperature:

>       description: Defines an action to run when the light is given a color temperature  command. Receives variable `color_temp`. May also receive variables `brightness`  and/or `transition`.

Reviewer: ```suggestion
      description: Defines an action to run when the light is given a color temperature command. Receives variable `color_temp`. May also receive variables `brightness`  and/or `transition`.
```

      required: false
      type: action
    set_hs:

>       description: "Defines an action to run when the light is given a hs colo  command. Available variables: `hs` as a tuple, `h` and `s`"

Reviewer: ```suggestion
      description: "Defines an action to run when the light is given a hs color command. Available variables: `hs` as a tuple, `h` and `s`"
```

      required: false
      type: action
    set_rgb:

>       description: "Defines an action to run when the light is given an RGB colo  command. Available variables: `rgb` as a tuple, `r`, `g` and `b`."

Reviewer: ```suggestion
      description: "Defines an action to run when the light is given an RGB color command. Available variables: `rgb` as a tuple, `r`, `g` and `b`."
```

      required: false
      type: action
    set_rgbw:

>       description: "Defines an action to run when the light is given an RGBW colo  command. Available variables: `rgbw` as a tuple, `rgb` as a tuple, `r`, `g`, `b`  and `w`."

Reviewer: ```suggestion
      description: "Defines an action to run when the light is given an RGBW color command. Available variables: `rgbw` as a tuple, `rgb` as a tuple, `r`, `g`, `b`  and `w`."
```

      required: false
      type: action
    set_rgbww:

>       description: "Defines an action to run when the light is given an RGBWW colo  command. Available variables: `rgbww` as a tuple, `rgb` as a tuple, `r`, `g`  `b`, `cw` and `ww`."

Reviewer: ```suggestion
      description: "Defines an action to run when the light is given an RGBWW color command. Available variables: `rgbww` as a tuple, `rgb` as a tuple, `r`, `g`  `b`, `cw` and `ww`."
```

      required: false
      type: action
    set_effect:

>       description: Defines an action to run when the light is given an effect command  Receives variable `effect`. May also receive variables `brightness` and/o  `transition`.

Reviewer: ```suggestion
      description: Defines an action to run when the light is given an effect command. Receives the variable `effect`. May also receive the variables `brightness`, and/or  `transition`.
```
